DESCRIPTION

```diff
@@ -29,7 +29,8 @@ Imports:
     stringi,
     stats,
     Rcpp (>= 0.12.7),
-    utils
+    utils,
+    purrr
```

Comment:
Can we do this without purrr? I'm hesitant to add any dependencies.

Reply:
For this simple ROW support, I think I can use `lappy()` in place of `purrr::map()`. However, I already have a  rough plan on how to make ARRAY/MAP/ROW support more comprehensive. In that plan, I use extensively functions from `purrr` as it is built to better deal with lists, especially nested lists (which are how the JSON objects of both the schema and data are).

I want to understand a bit better why you prefer not to add the dependency.
1) quality of the `purrr` package
2) increased size of RPresto
3) speed impact on functions in RPresto
4) something else

IMHO, `purrr` has become a quite fundamental part of the R and tidyverse ecosystem (literally 100+ CRAN packages now import it as a dependency, see https://cran.r-project.org/web/packages/purrr/index.html).

Reply:
My main concern is it historically had an unstable API. If we use core features that are stable (and unlikely to be deprecated on a whim in the future), it's fine.

Reply:
That's a valid concern when it comes to adding a new dependency. I've followed `purrr`'s development for 5+  years now. I think most functions in the package have been stable. The ones I used (i.e., `purrr::map_*()` and `purrr::modify_*()`) have been the core of `purrr` and should be kept stable even when a new version is released. I will refrain from using those with "questioning" lifecycle.

Reply:
Sounds good.